File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        pos, l, cnt = 0, list(), 0
        for i in range(len(nums)):
            if nums[i]<pivot:
                nums[pos], nums[i] = nums[i], nums[pos]
                pos+=1
            else:
                if nums[i] == pivot:
                    cnt+=1
                else:
                    l.append(nums[i])
        nums[pos:pos+cnt] = [pivot]*cnt
        pos+=cnt
        nums[pos::] = l
        return nums

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.js

const pivotArray = (nums, pivot) => {
	let N = nums.length;
	let output = [];

  // pointers to iterate array from start/end
	let i = 0;
	let j = N - 1;

  // pointers to save elements in correct spot
	let L = 0;
	let R = N - 1;
	
	// select elements starting from start and end of array
	// add elements to correct spot if conditions are met.
	while (i < nums.length) {
		if (nums[i] < pivot) output[L++] = nums[i];
		if (nums[j] > pivot) output[R--] = nums[j];

		i++;
		j--;
	}

	// fill middle with pivot
	while (L <= R) {
		output[L++] = pivot;
	}

	return output;
};